Modules/Core/Common/include/itkMersenneTwisterRandomVariateGenerator.h

```diff
@@ -296,12 +282,6 @@ class ITKCommon_EXPORT MersenneTwisterRandomVariateGenerator : public RandomVari
   static IntegerType
   GetNextSeed();
 
-  /*
-  // Saving and loading generator state
-  void save( IntegerType* saveArray ) const;  // to array of size SAVE
-  void load( IntegerType *const loadArray );  // from such array
-  */
```

Comment:
For the record, the original `MTRand` implementation _did_ implement `save` and `load`, at https://www.cs.rpi.edu/academics/courses/fall12/ds/hw/05_multi_linked_lists/MersenneTwister.h

```cpp
inline void MTRand::save( uint32* saveArray ) const
{
	register const uint32 *s = state;
	register uint32 *sa = saveArray;
	register int i = N;
	for( ; i--; *sa++ = *s++ ) {}
	*sa = left;
}

inline void MTRand::load( uint32 *const loadArray )
{
	register uint32 *s = state;
	register uint32 *la = loadArray;
	register int i = N;
	for( ; i--; *s++ = *la++ ) {}
	left = *la;
	pNext = &state[N-left];
}

```

However, they were never implemented in ITK's `MersenneTwisterRandomVariateGenerator`, so I guess they were not really necessary so far. 🤷

Reply:
I wonder if we should hove to the C++ STL Mersenne Twister? https://cplusplus.com/reference/random/mt19937/

Reply:
Oh, it is available since C++11. I wouldn't mind that being done, but I cannot devote resources towards that refactoring. @N-Dekker what about you?

Reply:
I think it would indeed be nice to move to the C++ STL Mersenne Twister. But initially I think we should then still preserve the interface of MersenneTwisterRandomVariateGenerator, and just have it implemented internally in terms of `std::mt19937`. The logic of having a global instance and a globally shared `m_StaticDiffer` is not in `std::mt19937`. But `std::mt19937` can certainly do the bit twisting magic for us  😃 As in our `MersenneTwisterRandomVariateGenerator::reload()`: https://github.com/InsightSoftwareConsortium/ITK/blob/7370ffd1f07c9f455f1dedad8d0dbc41c71bb7d5/Modules/Core/Common/src/itkMersenneTwisterRandomVariateGenerator.cxx#L212-L220

Reply:
That is a reasonable step towards using it directly.